src/hooks/index.tsx

```diff
@@ -118,3 +120,36 @@ export const useRefetchMinichefSubgraph = () => {
 
   return async () => await queryClient.refetchQueries(['get-minichef-farms-v2', account]);
 };
+
+export function useGetBlockTimestamp(blockNumber?: number) {
+  const latestBlockNumber = useBlockNumber();
+  const _blockNumber = blockNumber ?? latestBlockNumber;
+
+  const { connector } = useWeb3React();
+
+  const [timestamp, setTimestamp] = useState<BigNumber | undefined>(undefined);
+
+  useEffect(() => {
+    async function getTimeStamp() {
+      if (!_blockNumber) return;
+
+      const provider =
+        window.ethereum || window.bitkeep?.ethereum || window.xfi?.ethereum || (await connector?.getProvider());
+
+      const result: { timestamp: string } | null = await provider.request({
+        method: 'eth_getBlockByNumber',
+        params: [`0x${_blockNumber.toString(16)}`, false],
+      });
```

Comment:
this method needs to be added to `src/connectors/WalletProviders` to support this method in all chains. Ping me if you have any doubts